client/components/tinymce/plugins/media/plugin.jsx

```diff
@@ -415,6 +423,35 @@ function mediaButton( editor ) {
 		}
 	} );
 
+	editor.addButton( 'wp_img_edit', {
+		tooltip: i18n.translate( 'Edit', { context: 'verb' } ),
+		icon: 'dashicon dashicons-edit',
+		onclick: function() {
```

Comment:
I'd add in a line:
`classes: 'toolbar-segment-start',`

...such that the `Edit` button is "grouped" with the next button (as I mention in the other inline comment, I feel it's conceptually similar to `Delete`).